x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotRecoveryStateIntegrationTests.java

```diff
@@ -73,7 +82,98 @@ protected Settings nodeSettings(int nodeOrdinal) {
         return builder.build();
     }
 
+    public void testRecoveryStateDisplayStageIsFinalizeUntilPreWarmFinishes() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        final String firstDataNode = internalCluster().startDataOnlyNode();
+        final String fsRepoName = randomAlphaOfLength(10);
+        final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        final String restoredIndexName = randomBoolean() ? indexName : randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        final String snapshotName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+
+        createRepository(fsRepoName, "fs");
+
+        final Settings.Builder originalIndexSettings = Settings.builder();
+        originalIndexSettings.put(INDEX_SOFT_DELETES_SETTING.getKey(), true);
+        originalIndexSettings.put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1);
+
+        createAndPopulateIndex(indexName, originalIndexSettings);
+
+        createFullSnapshot(fsRepoName, snapshotName);
+
+        assertAcked(client().admin().indices().prepareDelete(indexName));
+
+        final CountDownLatch latch = fillPreWarmThreadPool(firstDataNode);
+
+        mountSnapshot(fsRepoName, snapshotName, indexName, restoredIndexName, Settings.EMPTY);
+
+        assertBusy(() -> {
+            final List<RecoveryState> recoveryStates = getRecoveryStates(restoredIndexName).filter(
+                recoveryState -> recoveryState.getRecoverySource().getType() == SNAPSHOT
+            ).collect(Collectors.toList());
+            assertThat(recoveryStates, hasSize(1));
+            final RecoveryState recoveryState = recoveryStates.get(0);
+            assertThat(recoveryState.getDisplayStage(), equalTo(RecoveryState.Stage.FINALIZE));
```

Comment:
Can we also assert that `getStage` has the expected value? Just to be sure we caught the right situation.